File: Back/AlkemyChallenge/MateriaViewModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AlkemyChallenge
{
    class MateriaViewModel
    {
        public int IdMateria { get; set; }
        public string nombre { get; set; }
        public string horario { get; set; }
        public ProfesorViewModel profesor { get; set; }
        public string cupoMax { get; set; }
        public string descripcion { get; set; }
        public string programa { get; set; }
    }
}

File: Front/src/app/models/materia-view-model.ts
import { ProfesorViewModel } from './profesor-view-model'

export class MateriaViewModel {
    public idMateria : number
    public nombre : string
    public horario : string
    public profesor : ProfesorViewModel
    public cupoMax : string
    public descripcion : string
    public programa : string
    
}
